app/src/pages/index.tsx

```diff
@@ -54,12 +52,14 @@ export default function Home(props: { users: User[] }) {
   );
 }
 
-export const getServerSideProps: GetServerSideProps = async () => {
-  const users = await fetchUsers(3, 0)
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const page = context.query.page != undefined ? Number(context.query?.page) : 0;
+  const users = await fetchUsers(10, page);
 
   return {
     props: {
       users,
+      page,
     },
   };
 };
```

Comment:
```suggestion
export const getServerSideProps: GetServerSideProps = async (context) => {
  const { page } = context.query
  const castedPage = Number.isNaN(Number(page)) ? 0 : Number(page)
  const users = await fetchUsers(10, castedPage);

  return {
    props: {
      users,
      page: castedPage,
    },
  };
};
```
こちらしっかりと格闘してくださったご様子が見受けられて感激しておるのですが、このような形にしないと、`?page=a`などが渡されたときに`page` variableの値が`NaN`になってしまいます！
また、このsuggestで自分か書いてみている、[分割代入](https://typescriptbook.jp/reference/values-types-variables/object/destructuring-assignment-from-objects)はよく使う文法ですので、覚えていただけると助かります！←てか覚えてますかねもう